runtime/src/snapshot_utils.rs

```diff
@@ -843,8 +843,19 @@ pub fn create_accounts_run_and_snapshot_dirs(
 ) -> std::io::Result<(PathBuf, PathBuf)> {
     let run_path = account_dir.as_ref().join("run");
     let snapshot_path = account_dir.as_ref().join("snapshot");
-    fs::create_dir_all(&run_path)?;
-    fs::create_dir_all(&snapshot_path)?;
+    if fs::metadata(&run_path).is_err() || fs::metadata(&snapshot_path).is_err() {
```

Comment:
It is OR so that it is done if any of run/ or snapshot/ is missing.

if run/ is missing, we should create run/ here.  AND will cause run/ creation to be skipped.

Reply:
I think OR is broader and safer.  If any of them is missing, it is considered an error condition, so cleanup and directory creation should be done.

If they both exist, no need to make the file system calls to recreate them.

Reply:
These directories are not supposed to be touched by the users.  If any unexpected directory change is detected, these directories should be considered as in an unknown state.  I feel there is no need to try to see which part of unexpected change could still be tolerated and handled differently.

Reply:
The current way is simple, only differentiating if it is the expected/known state, or unexpected/unknown state state (being unexpected could be older version, or user touching the directories).

If we really want to support the scenario of a user manually removes run/ but snapshot/ exists, then we need to guess how it came to this state.  was it from a new image, but user manually removed run/, or from an old image and a user manually added snapshot/?  should we check if there are appendvecs and then remove them?  when removing contents, we cannot call existing any function.  we have to code new logic to remove everything except snapshot/.

We can probably simplify the scenario with these assumption:
the user may remove run/, but he/she won't add snapshot/.
so if we see run/ missing, but snapshot/ exists, it must be from a new image, so there is no appendvec files to cleanup.  Then we can code in the suggested way: cleanup only in AND condition, and create the sub directories unconditionally.

Reply:
In the AND case, if user manually replace run/ with a run file, the code will silently move on, without cleanup, without creating run/.  right?